src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnectorConfig.java

```diff
@@ -815,4 +836,45 @@ public void ensureValid(String name, Object value) {
           this.validator.ensureValid(name, value);
         }
       };
+
+  /**
+   * Class which validates key value pairs in the format <key-1>:<value-1>,<key-2>:<value-2>
+   *
+   * <p>It doesn't validate the type of values, only making sure the format is correct.
+   */
+  public static class CommaSeparatedKeyValueValidator implements ConfigDef.Validator {
+    public CommaSeparatedKeyValueValidator() {}
+
+    public void ensureValid(String name, Object value) {
+      String s = (String) value;
+      // Validate the comma-separated key-value pairs string
+      if (s != null && !s.isEmpty() && !isValidCommaSeparatedKeyValueString(s)) {
```

Comment:
well it can, but this order signifies that it can be null or empty. we want to check only if its non null and non empty. i prefer it here for readability.